src/os/bluestore/BlueStore.cc

```diff
@@ -1189,6 +1189,8 @@ void BlueStore::SharedBlob::put()
 	dout(20) << __func__ << " " << this
 		 << " lost race to remove myself from set" << dendl;
       }
+    } else {
+      delete this;
```

Comment:
@liewegas This is causing the following crash.

/root/ceph-master/src/os/bluestore/BlueStore.h: 319: FAILED assert(bc.empty())

 ceph version v11.0.0-2254-g8c52058 (8c52058bc8a0f060e06037196627c19b7d258f1f)
 1: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x80) [0x55d6cde26ef0]
 2: (BlueStore::SharedBlob::~SharedBlob()+0x82) [0x55d6cdbabee2]
 3: (BlueStore::SharedBlob::put()+0x17d) [0x55d6cdb581dd]
 4: (BlueStore::Onode::put()+0x1e8) [0x55d6cdbaca88]
 5: (BlueStore::TwoQCache::trim(unsigned long, unsigned long)+0x7b3) [0x55d6cdb6e6d3]
 6: (BlueStore::getattr(boost::intrusive_ptr<ObjectStore::CollectionImpl>&, ghobject_t const&, char const*, ceph::buffer::ptr&)+0x120) [0x55d6cdb93810]
 7: (PGBackend::objects_get_attr(hobject_t const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, ceph::buffer::list*)+0x106) [0x55d6cd970216]
 8: (ReplicatedPG::get_object_context(hobject_t const&, bool, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, ceph::buffer::list, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, ceph::buffer::list> > >*)+0x92e) [0x55d6cd8f0cee]
 9: (ReplicatedPG::prep_object_replica_pushes(hobject_t const&, eversion_t, PGBackend::RecoveryHandle*)+0x1a4) [0x55d6cd8f9654]
 10: (ReplicatedPG::recover_replicas(unsigned long, ThreadPool::TPHandle&)+0x1093) [0x55d6cd937d83]
 11: (ReplicatedPG::start_recovery_ops(unsigned long, ThreadPool::TPHandle&, unsigned long*)+0xd86) [0x55d6cd9406a6]
 12: (OSD::do_recovery(PG*, unsigned int, unsigned long, ThreadPool::TPHandle&)+0x426) [0x55d6cd7b2f96]
 13: (OSD::ShardedOpWQ::_process(unsigned int, ceph::heartbeat_handle_d*)+0x5bf) [0x55d6cd7b3b6f]
 14: (ShardedThreadPool::shardedthreadpool_worker(unsigned int)+0x7f9) [0x55d6cde2bca9]
 15: (ShardedThreadPool::WorkThreadSharded::entry()+0x10) [0x55d6cde2f060]
 16: (()+0x76fa) [0x7fd9251296fa]
 17: (clone()+0x6d) [0x7fd923a14b5d]